homeassistant/components/group/media_player.py

```diff
@@ -0,0 +1,405 @@
+
+from typing import Any, Callable
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    ATTR_MEDIA_CONTENT_ID,
+    ATTR_MEDIA_CONTENT_TYPE,
+    ATTR_MEDIA_SEEK_POSITION,
+    ATTR_MEDIA_SHUFFLE,
+    ATTR_MEDIA_VOLUME_LEVEL,
+    ATTR_MEDIA_VOLUME_MUTED,
+    DOMAIN,
+    PLATFORM_SCHEMA,
+    SERVICE_CLEAR_PLAYLIST,
+    SERVICE_MEDIA_NEXT_TRACK,
+    SERVICE_MEDIA_PAUSE,
+    SERVICE_MEDIA_PLAY,
+    SERVICE_MEDIA_PREVIOUS_TRACK,
+    SERVICE_MEDIA_SEEK,
+    SERVICE_MEDIA_STOP,
+    SERVICE_PLAY_MEDIA,
+    SERVICE_SHUFFLE_SET,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    SERVICE_VOLUME_MUTE,
+    SERVICE_VOLUME_SET,
+    SUPPORT_CLEAR_PLAYLIST,
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PLAY_MEDIA,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_SEEK,
+    SUPPORT_SHUFFLE_SET,
+    SUPPORT_STOP,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+    MediaPlayerEntity,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
+    CONF_ENTITIES,
+    CONF_NAME,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import HomeAssistant, State
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType, EventType
+
+KEY_CLEAR_PLAYLIST = "clear_playlist"
+KEY_ON_OFF = "on_off"
+KEY_PAUSE_PLAY_STOP = "play"
+KEY_PLAY_MEDIA = "play_media"
+KEY_SHUFFLE = "shuffle"
+KEY_SEEK = "seek"
+KEY_TRACKS = "tracks"
+KEY_VOLUME = "volume"
+
+DEFAULT_NAME = "Media Group"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_ENTITIES): cv.entities_domain(DOMAIN),
+    }
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: Callable,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Media Group platform."""
+    async_add_entities([MediaGroup(config[CONF_NAME], config[CONF_ENTITIES])])
+
+
+class MediaGroup(MediaPlayerEntity):
+    """Representation of a Media Group."""
+
+    def __init__(self, name: str, entities: list[str]) -> None:
+        """Initialize a Media Group entity."""
+        self._name = name
+        self._state: str | None = None
+        self._supported_features: int = 0
+
+        self._entities = entities
+        self._features: dict[str, set[str]] = {
+            KEY_CLEAR_PLAYLIST: set(),
+            KEY_ON_OFF: set(),
+            KEY_PAUSE_PLAY_STOP: set(),
+            KEY_PLAY_MEDIA: set(),
+            KEY_SHUFFLE: set(),
+            KEY_SEEK: set(),
+            KEY_TRACKS: set(),
+            KEY_VOLUME: set(),
+        }
+
+    async def _on_state_change(self, event: EventType) -> None:
```

Comment:
And then we can make this a callback too. Only some standard methods are required to be either sync functions to coroutine functions. This is documented in our Entity developer docs.

Reply:
Done